Address linter errors

`golangci-lint` was updated to [v1.46.0](https://github.com/golangci/golangci-lint/releases/tag/v1.46.0) — with support for Go 1.18 among other things — and the following errors are now returned on `main`:

```
  Error: pkg/targets/adapter/test/test_cloudevents.go:119:2: S1001: should use copy() instead of a loop (gosimple)
  	for i := range c.sent {
  	^
  Error: pkg/sources/adapter/httppollersource/adapter.go:28:2: ST1019: package "knative.dev/eventing/pkg/adapter/v2" is being imported more than once (stylecheck)
  	"knative.dev/eventing/pkg/adapter/v2"
  	^
  Error: pkg/sources/adapter/httppollersource/adapter.go:29:2: ST1019(related information): other import of "knative.dev/eventing/pkg/adapter/v2" (stylecheck)
  	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
  	^
  Error: pkg/sources/adapter/httppollersource/httppoller.go:30:2: ST1019: package "knative.dev/eventing/pkg/adapter/v2" is being imported more than once (stylecheck)
  	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
  	^
  Error: pkg/sources/adapter/httppollersource/httppoller.go:32:2: ST1019(related information): other import of "knative.dev/eventing/pkg/adapter/v2" (stylecheck)
  	"knative.dev/eventing/pkg/adapter/v2"
  	^
```

--- a/pkg/sources/adapter/httppollersource/adapter.go
+++ b/pkg/sources/adapter/httppollersource/adapter.go
@@ -25,7 +25,6 @@ import (
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 	"go.uber.org/zap"
 
-	"knative.dev/eventing/pkg/adapter/v2"
 	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
 	"knative.dev/pkg/logging"
 
@@ -33,7 +32,7 @@ import (
 )
 
 // NewAdapter satisfies pkgadapter.AdapterConstructor.
-func NewAdapter(ctx context.Context, envAcc adapter.EnvConfigAccessor, ceClient cloudevents.Client) adapter.Adapter {
+func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
 	logger := logging.FromContext(ctx)
 
 	mt := &pkgadapter.MetricTag{

--- a/pkg/sources/adapter/httppollersource/httppoller.go
+++ b/pkg/sources/adapter/httppollersource/httppoller.go
@@ -28,8 +28,6 @@ import (
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 
 	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
-
-	"knative.dev/eventing/pkg/adapter/v2"
 )
 
 type httpPoller struct {
@@ -45,7 +43,7 @@ type httpPoller struct {
 	mt          *pkgadapter.MetricTag
 }
 
-var _ adapter.Adapter = (*httpPoller)(nil)
+var _ pkgadapter.Adapter = (*httpPoller)(nil)
 
 // Start implements adapter.Adapter.
 // Runs the server for receiving HTTP events until ctx gets cancelled.

--- a/pkg/targets/adapter/test/test_cloudevents.go
+++ b/pkg/targets/adapter/test/test_cloudevents.go
@@ -116,9 +116,7 @@ func (c *FakeCloudEventsClient) Sent() []cloudevents.Event {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	r := make([]cloudevents.Event, len(c.sent))
-	for i := range c.sent {
-		r[i] = c.sent[i]
-	}
+	copy(r, c.sent)
 	return r
 }
 

